src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -259,19 +260,31 @@ public static void AddScale(float a, ReadOnlySpan<float> src, ReadOnlySpan<int>
 
             if (Avx.IsSupported)
             {
-                AvxIntrinsics.AddScaleSU(a, src, indices, dst, count);
+                if (VectorizationMakesSense(count))
+                    AvxIntrinsics.AddScaleSU(a, src, indices, dst, count);
+                else
+                    AddScaleSimple(a, src, indices, dst, count);
             }
             else if (Sse.IsSupported)
             {
-                SseIntrinsics.AddScaleSU(a, src, indices, dst, count);
+                if (VectorizationMakesSense(count))
+                    SseIntrinsics.AddScaleSU(a, src, indices, dst, count);
+                else
+                    AddScaleSimple(a, src, indices, dst, count);
             }
             else
             {
-                for (int i = 0; i < count; i++)
-                {
-                    int index = indices[i];
-                    dst[index] += a * src[i];
-                }
+                AddScaleSimple(a, src, indices, dst, count);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void AddScaleSimple(float a, ReadOnlySpan<float> src, ReadOnlySpan<int> indices, Span<float> dst, int count)
```

Comment:
(minor) It feels odd to use two different terms: `Vectorization` and `Simple` in this code. Can we stick with one term?  Maybe `AddScaleNonvectorized`?